commands/rank.js

```diff
@@ -0,0 +1,131 @@
+const Discord = require('discord.js');
+const Enmap = require("enmap");
+
+const ALLOWED_RANKS = [
+  "Iron1",
+  "Iron2",
+  "Iron3",
+  "Bronze1",
+  "Bronze2",
+  "Bronze3",
+  "Silver1",
+  "Silver2",
+  "Silver3",
+  "Gold1",
+  "Gold2",
+  "Gold3",
+  "Platinum1",
+  "Platinum2",
+  "Platinum3",
+  "Diamond1",
+  "Diamond2",
+  "Diamond3",
+  "Immortal1",
+  "Immortal2",
+  "Immortal3",
+  "Radiant"
+];
+
+exports.run = async (client, message, args, level) => {
+  client.ranks = new Enmap({name: "ranks"});
+  const channel = message.channel;
+
+
+  if(args.length == 0) {
+    return sendHelp(channel);
+  }
+
+  switch(args[0]) {
+    case "set":
+      const rank = args[1];
+
+      if(!rank || !ALLOWED_RANKS.includes(rank)) {
+        // invalid rank, send allowed ranks
+        return sendRanks(channel);
+      } else {
+        const key = message.author.id;
+        const username = message.author.username;
+
+        // save rank
+        client.ranks.set(key, {
+          username: username,
```

Comment:
It might be better to only store the user id as the key and then retrieve the username (in case they changed it, it wouldn't be updated until next set). 

I think the user.cache fetching is pretty performant since its all in memory. 
I gave this a try in eval and it was not bad - `client.users.cache.get(id)`

Reply:
coo I didn't know this was possible

Reply:
I only know cause v12 apis broke the bot and I had to google how to migrate some of the v11 commands to v12 on the template :)